connection.go

```diff
@@ -1163,3 +1176,54 @@ func (conn *Connection) NewStream() (*Stream, error) {
 		Conn: conn,
 	}, nil
 }
+
+// loadProtocolInfo sends info about client protocol,
+// receives info about server protocol in response
+// and store in in connection serverProtocolInfo.
+func (conn *Connection) loadProtocolInfo() error {
+	var ok bool
+
+	resp, err := conn.exchangeProtocolInfo(
+		ClientProtocolVersion,
+		ClientProtocolFeatures)
+
+	if err != nil {
+		if resp.Code == ErrUnknownRequestType {
+			// IPROTO_ID requests are not supported by server.
+			conn.serverProtocolInfo = protocolInfo{
+				version:  ProtocolVersionUnsupported,
+				features: []ProtocolFeature{},
+			}
+			return nil
+		}
+
+		return err
+	}
+
+	if len(resp.Data) == 0 {
+		return fmt.Errorf("Unexpected response on protocol info exchange: no data")
+	}
+
+	conn.serverProtocolInfo, ok = resp.Data[0].(protocolInfo)
+	if !ok {
+		return fmt.Errorf("Unexpected response on protocol info exchange: wrong data")
+	}
+
+	return nil
+}
+
+// IsProtocolVersionSupported checks if expected protocol version
+// is supported by both server and client.
+// Since 1.10.0
+func (conn *Connection) IsProtocolVersionSupported(version ProtocolVersion) bool {
+	return (version <= ClientProtocolVersion) &&
+		(version <= conn.serverProtocolInfo.version)
+}
+
+// IsProtocolFeatureSupported checks if expected protocol feature
+// is supported by both server and client.
+// Since 1.10.0
+func (conn *Connection) IsProtocolFeatureSupported(feature ProtocolFeature) bool {
+	return isFeatureSupported(feature, ClientProtocolFeatures) &&
+		isFeatureSupported(feature, conn.serverProtocolInfo.features)
+}
```

Comment:
This interface hides potentially useful information:

1. Supported protocol version by a server.
2. Supported features by a server.

For example, now it is impossible to check is the server supports MP_ERROR or watchers.

Reply:
How this info could be useful to a user?

Reply:
Well, feature flags exist to display if some codes are known (or unknown) to IProto. I'm not sure they are suitable to check Tarantool features disregarding using any client.

I see the reason to write a code like this:
```
if (server_support_watchers and client_support_watchers) {
    -- code with watchers
} else {
    -- code without watchers
}
```

Reply:
I would still not try to guess the method a use case because we may be wrong.

We have an information about server, let's clearly share it.

Reply:
Well, ok, that's fair